src/sys/unix/epoll.rs

```diff
@@ -171,6 +168,69 @@ impl Drop for Selector {
     }
 }
 
+pub type RawEvent = libc::epoll_event;
+
+#[repr(transparent)]
+pub struct Event {
+    inner: RawEvent,
+}
+
+impl Event {
+    pub fn token(&self) -> Token {
+        Token(self.inner.u64 as usize)
+    }
+
+    pub fn is_readable(&self) -> bool {
+        (self.inner.events as libc::c_int & libc::EPOLLIN) != 0
+            || (self.inner.events as libc::c_int & libc::EPOLLPRI) != 0
+    }
+
+    pub fn is_writable(&self) -> bool {
+        (self.inner.events as libc::c_int & libc::EPOLLOUT) != 0
+    }
+
+    pub fn is_error(&self) -> bool {
+        (self.inner.events as libc::c_int & libc::EPOLLERR) != 0
+    }
+
+    pub fn is_hup(&self) -> bool {
+        (self.inner.events as libc::c_int & libc::EPOLLHUP) != 0
+            || (self.inner.events as libc::c_int & libc::EPOLLRDHUP) != 0
+    }
+
+    pub fn is_priority(&self) -> bool {
+        (self.inner.events as libc::c_int & libc::EPOLLPRI) != 0
+    }
+
+    pub fn is_aio(&self) -> bool {
+        // Not supported in the kernel, only in libc.
+        false
+    }
+
+    pub fn is_lio(&self) -> bool {
+        // Not supported.
+        false
+    }
+
+    pub fn raw_event(&self) -> &RawEvent {
+        &self.inner
+    }
+
+    pub fn from_raw_event(epoll_event: RawEvent) -> Event {
+        Event { inner: epoll_event }
+    }
+}
+
+impl fmt::Debug for Event {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // TODO: add readiness.
+        f.debug_struct("Event")
+            .field("token", &self.token())
+            //.field("readiness", &self.readiness())
```

Comment:
Can we do with readiness? I could create a function `epoll_event`/`kevent` -> `Ready`, but is worth the effort?

Reply:
I've moved the `Debug` impl to `event_imp.rs`.